### ISSUE TITLE
Pushing directly to Master

### ISSUE THREAD

**Issue by itdependsnetworks** [CiscoTestAutomation/genieparser#244]
Is this practice to be expected? 

I see the following issues with this practice
* People watching the repo will not be notified
* No "proof" that tests passed before being updated
  * Seems to be likely the alternate workflow is causing tests to not be ran in the same way, as I have seen cases where tests should have failed, but yet were in master. (starting to make sense now)
  * The tests that have been added are incorrect, as they did not whitelist the folder based tests, and seems to be moving backwards
* Seems to be a difference between internal processes and external. 
  * make json does not work (unclear of the intention)
  * runall is only provided locally
* Potential issue to force merge into master.
* In a spacial system such as regex, there is consistent incorrect raw output provided, see below as an example taken from a direct merge sent today:

```
    golden_output = {'execute.return_value': '''
        pm9010#show sdwan utd engine
        utd-oper-data utd-engine-status version 1.0.6_SV2.9.13.0_XE17.3
        utd-oper-data utd-engine-status profile Cloud-Medium
        utd-oper-data utd-engine-status status utd-oper-status-green
        utd-oper-data utd-engine-status reason ""
        utd-oper-data utd-engine-status memory-usage 11.3
        utd-oper-data utd-engine-status memory-status utd-oper-status-green
        ID  RUNNING  STATUS                 REASON  
        --------------------------------------------
        1   true     utd-oper-status-green       
        2   true     utd-oper-status-green      
        '''
        }
```

pinging @ahersi2-cisco https://github.com/CiscoTestAutomation/genieparser/commit/35471076b60e2623ce3b860b92f0702fc1de4a8d

**Comment by jeaubin** [comment]
To give you a better understanding of the internal workflow - 

Internally we are using Github enterprise; following same guidelines as here - make sure unittest pass, etc. (Github enterprise doesnt support workflow yet, so we are using jenkins). Once a month at the release time - or close by, we take all that was worked during that month and merge it here. 


* We should be creating a new PR for our monthly merges,

* Im suprised make json doesnt work - we will fix

* runall - yes thats annoying; we need to release it externally, will get on it 

* For the space,  let me make sure that doesnt happen.

* We will switch to the new unittest structure / no PR approve without it starting next week.  



I also want to iterate,  thank you for pushing us towards these changes, I appreciate it; we need to switch more and more towards a more open source model - instead of the cisco model we've been doing for years.

**Comment by itdependsnetworks** [comment]
Thanks.

Is it possible you are not pulling down from the open source project? There seems to be a gap and that is my only guess.

**Comment by jeaubin** [comment]
Can you explain what you mean by this ? What is the gap ?

**Comment by itdependsnetworks** [comment]
Tests are not passing when as an open source contributor I rebase from master.

**Comment by mgdransfield** [comment]
Is this causing make json to fail as iosxe/show_wireless.py has multiple errors in it?

**Comment by LenzDong** [comment]
Hi,

We have released latest v20.10 last week. And I don't see the issue with latest release.
I will close this thread. Please update with the latest version. Let us know if any other issue with latest.

Thanks,
